cryptographic_estimators/MQEstimator/series/nmonomial.py

```diff
@@ -40,55 +39,29 @@ class NMonomialSeries(object):
     """
 
     def __init__(self, n: int, q=None, max_prec=None):
+
         self._n = n
         if max_prec is not None:
             self._max_prec = max_prec
         else:
             self._max_prec = self._n + 1
 
-        R = PowerSeriesRing(QQ, 'z', default_prec=self._max_prec)
-        z = R.gen()
+        x = power_series([0, 1], prec=self._max_prec)
 
         if q is not None:
             if not is_prime_power(q):
-                raise ValueError(
-                    "the order of finite field q must be a prime power")
+                raise ValueError("the order of finite field q must be a prime power")
             self._q = q
             if q < self._max_prec:
-                self._series_of_degree = (
-                    (1 - z ** self._q) ** self._n) / ((1 - z) ** self._n)
+                self._series_of_degree = ((1 - x**self._q) ** self._n) / (
+                    (1 - x) ** self._n
+                )
             else:
-                self._series_of_degree = 1 / ((1 - z) ** self._n)
+                self._series_of_degree = 1 / ((1 - x) ** self._n)
         else:
-            self._series_of_degree = 1 / ((1 - z) ** self._n)
-
-        self._series_up_to_degree = self._series_of_degree / (1 - z)
-
-    def series_monomials_of_degree(self):
-        """
-        Return the series of the number of monomials of a given degree
-
-        EXAMPLES::
+            self._series_of_degree = 1 / ((1 - x) ** self._n)
```

Comment:
Don't we have methods in flint to print the series?

Reply:
... I don't get why the question, but yes, we have two ways:
```python
>>> x = flint.fmpz_series([0,1])

>>> x.repr()
'fmpz_series([0, 1], prec=10)'

>>> print(x)
x + O(x^10)
```

Do you want some method to expose any of those?

Reply:
I was wondering if it is possible to keep test or examples like following

` EXAMPLES::

            sage: from cryptographic_estimators.MQEstimator.series.nmonomial import NMonomialSeries
            sage: NM = NMonomialSeries(n=6, q=5)
            sage: NM.series_monomials_of_degree()
            1 + 6*z + 21*z^2 + 56*z^3 + 126*z^4 + 246*z^5 + 426*z^6 + O(z^7)`
``

Reply:
While this is in general a bad practice, I also think it could be good to keep methods like this for testing and development process. Please look my last commit. 

Methods are private as they must not be used in other files.